examples/audio/audio_looping.rs

```diff
@@ -0,0 +1,46 @@
+use bevy::prelude::*;
+
+/// This example illustrates how to load and play an audio file on repeat.
+
+struct OverworldThemeTimer(Timer);
+
+struct OverworldTheme {
+    track: Handle<AudioSource>,
+}
+
+const THEME_LENGTH: f32 = 13.;
```

Comment:
Is there any way to grab this metadata from the audio sample itself? Magic numbers like this are going to be very brittle.

Reply:
This one's got me scratching my head some, but here's where I'm at. 

When I load the audio asset with `asset_server.load("sounds/loop_me_melancholy.mp3")` I end up with a `Handle<AudioSource>`. And if I can get at that `AudioSource` itself, I think the track's duration is _almost_ available through the underlying `rodio::decoder::Source`. I could feasibly add an `impl` for `AudioSource` in a separate PR that does something like
```rust
impl AudioSource {
    pub fn get_track_duration(&self) -> Duration {
        self.get_track_duration()
    }
}
```
but can't actually get at rodio's `get_track_duration()` in my example w/o some acrobatics because the `rodio` crate's traits aren't in scope there.

All that aside 😅... I can't quite figure a good way to get the `AudioSource` from my `Handle<AudioSource>` in a startup system or that `FromWorld` trait 🙃 .

I do think I can get at the `AudioSource` after initialization and startup, in a system that uses a parameter like `Res<Assets<AudioSource>>` but by that point it feels like I've ended up defeating the purpose of using the metadata to initialize a timer from the get go.

I'm going to keep chipping at it but any pointers you have would be appreciated 😄 🙏

Reply:
>  I could feasibly add an impl for AudioSource in a separate PR that does something like

This sounds like a great choice.

> 

> I can't quite figure a good way to get the AudioSource from my Handle<AudioSource> in a startup system or that FromWorld trait 🙃.

You need to access it through the `Assets` resource :) https://docs.rs/bevy/0.5.0/bevy/asset/struct.Assets.html#method.get

Within the `FromWorld` trait, you can grab the resource in question off the `World`, and from there it should be straightforward :)

Reply:
Thanks for the reply and advice @alice-i-cecile 🙌 🙌 🙏 🙏 

Unfortunately it hasn't been super straight forward for me to get the `AudioSource` I need from the `Assets` resource because I have to figure a way to wait for it to load... I think. 

In `examples/asset/asset_loading.rs` there's something detailing pretty much my exact scenario here https://github.com/bevyengine/bevy/blob/54ff7aaa1e1c0edaf8885a601bdb1ab29a90d3dd/examples/asset/asset_loading.rs#L27-L37

It it easy to wait around for an asset to load? Right now I'm doing things in `FromWorld` that I'm embarrassed to admit, like making `AssetServer::load_async` public and using `futures_lite::future::block_on` in my example (especially embarrassed because none of it's actually working 😆)

Reply:
> Unfortunately it hasn't been super straight forward for me to get the AudioSource I need from the Assets resource because I have to figure a way to wait for it to load... I think.

Correct :)

I'm not thrilled with how easy it is to wait for assets to load: see #1701 for some templates and discussion on how we might improve this in the future <3 Thanks a ton for working on this; hopefully it's been educational!